src/main/scala/com/fulcrumgenomics/fasta/CollectAlternateContigNames.scala

```diff
@@ -192,11 +210,22 @@ class CollectAlternateContigNames
         }
       }
     }
-    // Sort the secondary entries based on descending sequence length.
-    val infos = (primaries ++ secondaries.sortBy(-_.length))
-      .zipWithIndex
-      .map { case (metadata, index) => metadata.copy(index=index) }
-      .toSeq
+
+    // Get the ordering of the molecules
+    val infos: Seq[SequenceMetadata] = existingSequenceDictionary match {
+      case None =>
+        // Sort the secondary entries based on descending sequence length.
+        (primaries ++ secondaries.sortBy(-_.length))
+          .zipWithIndex
+          .map { case (metadata, index) => metadata.copy(index=index) }
+          .toSeq
+      case Some(sequenceDictionary) =>
```

Comment:
I think this needs to be done differently.  The docs say "update an existing sequence dictionary" but really this is just "ordering by an existing sequence dictionary" and will drop any tags (e.g. MD5) not generated by `CollectAlternateContigNames`.  I think this should be implemented to cycle through the input dictionary and generate copies of the _input_ metadatas that:

1. Add any new aliases from the report that are not already contained in the input record
2. Add/update any other tags extended tags on the input record that are defined on the new record